Mage.Sets/src/mage/cards/e/ElrondOfTheWhiteCouncil.java

```diff
@@ -0,0 +1,186 @@
+import mage.util.CardUtil;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * @author Susucr
+ */
+public final class ElrondOfTheWhiteCouncil extends CardImpl {
+
+    public ElrondOfTheWhiteCouncil(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{G}{U}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.ELF);
+        this.subtype.add(SubType.NOBLE);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Secret council -- When Elrond of the White Council enters the battlefield, each player secretly votes for fellowship or aid, then those votes are revealed. For each fellowship vote, the voter chooses a creature they control. You gain control of each creature chosen this way, and they gain "This creature can't attack its owner." Then for each aid vote, put a +1/+1 counter on each creature you control.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(
+                new ElrondOfWhiteCouncilEffect()
+        ).setAbilityWord(AbilityWord.SECRET_COUNCIL));
+    }
+
+    private ElrondOfTheWhiteCouncil(final ElrondOfTheWhiteCouncil card) {
+        super(card);
+    }
+
+    @Override
+    public ElrondOfTheWhiteCouncil copy() {
+        return new ElrondOfTheWhiteCouncil(this);
+    }
+}
+
+class ElrondOfWhiteCouncilEffect extends OneShotEffect {
+
+    ElrondOfWhiteCouncilEffect() {
+        super(Outcome.Benefit);
+        staticText = "each player secretly votes for fellowship or aid, then those votes are revealed. "
+                + "For each fellowship vote, the voter chooses a creature they control. You gain control "
+                + "of each creature chosen this way, and they gain \"This creature can't attack its owner.\" "
+                + "Then for each aid vote, put a +1/+1 counter on each creature you control.";
+    }
+
+    private ElrondOfWhiteCouncilEffect(final ElrondOfWhiteCouncilEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ElrondOfWhiteCouncilEffect copy() {
+        return new ElrondOfWhiteCouncilEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        TwoChoiceVote vote = new TwoChoiceVote(
+                "Fellowship",
+                "Aid",
+                Outcome.Benefit,
+                true
+        );
+        vote.doVotes(source, game);
+
+
+        Set<Permanent> chosenCreatures = new HashSet<>();
+
+        // For each fellowship vote,
+        for (UUID playerId : game.getState().getPlayersInRange(source.getControllerId(), game)) {
+            int countFellowship = (int) vote.getVotes(playerId).stream().filter(b -> b).count();
```

Comment:
Curious- why are we filtering from b to b, and why is the cast needed?

Reply:
count() returns a long.
The filter is that the votes are true for fellowship, and false for aid, and i wanted only the fellowship ones.

Reply:
Thanks for explaining, I get it now.